packages/amplify-category-function/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.js

```diff
@@ -384,43 +384,47 @@ async function askExecRolePermissionsQuestions(context, allDefaultValues, parame
     }
 
     try {
-      const { getPermissionPolicies } = require(categoryPlugins[category]);
-      if (!getPermissionPolicies) {
-        context.print.warning(`Policies cannot be added for ${category}`);
-        continue;
+      let selectedResources = [];
+
+      if (resourcesList.length === 1) {
+        context.print.info(`${capitalizeFirstLetter(category)} category has a resource called ${resourcesList[0]}`);
+        selectedResources = [resourcesList[0]];
       } else {
-        let selectedResources = [];
+        const resourceQuestion = {
+          type: 'checkbox',
+          name: 'resources',
+          message: `${capitalizeFirstLetter(category)} has ${
+            resourcesList.length
+          } resources in this project. Select the one you would like your Lambda to access`,
+          choices: resourcesList,
+          validate: value => {
+            if (value.length === 0) {
+              return 'You must select at least resource';
+            }
+            return true;
+          },
+          default: () => {
+            if (currentDefaults && currentDefaults.categoryPermissionMap && currentDefaults.categoryPermissionMap[category]) {
+              return Object.keys(currentDefaults.categoryPermissionMap[category]);
+            }
+          },
+        };
 
-        if (resourcesList.length === 1) {
-          context.print.info(`${capitalizeFirstLetter(category)} category has a resource called ${resourcesList[0]}`);
-          selectedResources = [resourcesList[0]];
-        } else {
-          const resourceQuestion = {
-            type: 'checkbox',
-            name: 'resources',
-            message: `${capitalizeFirstLetter(category)} has ${
-              resourcesList.length
-            } resources in this project. Select the one you would like your Lambda to access`,
-            choices: resourcesList,
-            validate: value => {
-              if (value.length === 0) {
-                return 'You must select at least resource';
-              }
-              return true;
-            },
-            default: () => {
-              if (currentDefaults && currentDefaults.categoryPermissionMap && currentDefaults.categoryPermissionMap[category]) {
-                return Object.keys(currentDefaults.categoryPermissionMap[category]);
-              }
-            },
-          };
+        const resourceAnswer = await inquirer.prompt([resourceQuestion]);
+        selectedResources = resourceAnswer.resources;
+      }
 
-          const resourceAnswer = await inquirer.prompt([resourceQuestion]);
-          selectedResources = resourceAnswer.resources;
-        }
+      for (let j = 0; j < selectedResources.length; j += 1) {
+        const resourceName = selectedResources[j];
 
-        for (let j = 0; j < selectedResources.length; j += 1) {
-          const resourceName = selectedResources[j];
+        const pluginInfo = context.amplify.getCategoryPluginInfo(context, category, resourceName);
```

Comment:
Is this the only major change? What else changed? Really hard to figure out with the diff.

Reply:
the change is that previously we didn't handle two category plugins having the same name, and we just return the first one in the array, but now, we distinguish them using the services they provide.